### PR TITLE
🛠️ Provide CORS protection against CSRF attacks 

### LINKED ISSUE #172: 🛠️ Provide CORS protection against CSRF attacks
Provide CORS protection to protect the app against external CSRF attacks by using the `actix-cors` crate.

### PR DESCRIPTION
**Pull request by neon-mmd**
## What does this PR do?

This PR implements CORS protection in the app.

## Why is this change important?

This change is essential as by implementing CORS in the app, it protects it against CSRF attacks.

## How to test this PR locally?

It can be tested by installing and running `Websurfx` as mentioned in the `docs` and on the `readme` and by launching the browser and thoroughly testing. By requesting data from the app using curl. For example:

``` bash
 curl -X "GET" -H "Content-Type: application/json" \
                       -H "Origin: http://127.0.0.1" \
                       -H "Access-Control-Request-Method: GET" \
                        -H "Access-Control-Request-Headers: X-Requested-With" \
                       "http://127.0.0.1:8080/settings" -v
```

## Author's checklist

- [x] add code to mitigate csrf vulnerability using cors.
- [x] bump the app version to `v0.16.1`

## Related issues

Closes #172

### CODE REVIEW

**Comment by xffxff** [comment]
I don't have much knowledge about CORS right now, but I will review it later when I have learned more and feel comfortable to do so

File: src/lib.rs [hunk 1]
```diff
@@ -50,9 +51,20 @@ pub fn run(listener: TcpListener, config: Config) -> std::io::Result<Server> {
     let handlebars_ref: web::Data<Handlebars> = web::Data::new(handlebars);
 
     let server = HttpServer::new(move || {
+        let cors: Cors = Cors::default()
+            .allow_any_origin()
+            .allowed_methods(vec!["GET"])
```
Comment (xffxff) [inline src/lib.rs:58]: Does this mean we allow any origin to send a GET request to our server? The request can be sent from any other origin, such as http://foo.bar.com

Comment (AapoSoukkio) [reply src/lib.rs:58]: Yes, that's correct. The provided code allows any origin to send a GET request to your server but if I have understand the use case of this project properly that's is not necessarily bad thing.

Comment (xffxff) [reply src/lib.rs:58]: @AapoSoukkio Thanks for explaining it.

**Review by xffxff — DISMISSED** [review]
After learning something about CORS, the changes now make sense to me

**Comment by xffxff** [comment]
Some conflicts need to be resolved @neon-mmd

**Comment by neon-mmd** [comment]
> Some conflicts need to be resolved @neon-mmd

Yes, sure :slightly_smiling_face: I fixed them. Make sure take a review on this PR again :slightly_smiling_face: .

**Review by xffxff — APPROVED** [review]

**Pull request merged**